coderd/templates_test.go

```diff
@@ -543,6 +604,96 @@ func TestPatchTemplateMeta(t *testing.T) {
 		})
 	})
 
+	t.Run("AllowUserScheduling", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("OK", func(t *testing.T) {
+			t.Parallel()
+
+			var (
+				setCalled      int64
+				allowAutoStart atomic.Bool
+				allowAutoStop  atomic.Bool
```

Comment:
When I initially looked at this, I was wondering whether it would make more sense to instead use a `chan bool` instead to assert the values passed to `setFn`. But this does seem to be simpler in some ways and, crucially, will fail _quickly_ if it fails. Not so for a channel... so 👍